.buildkite/commands/run-swiftlint.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -u
+
+echo "--- :swift: Running SwiftLint"
+
+set +e
+SWIFTLINT_OUTPUT=$(swiftlint lint --quiet $@ --reporter csv)
+SWIFTLINT_EXIT_STATUS=$?
+set -e
+
+WARNINGS=$(echo "$SWIFTLINT_OUTPUT" | awk -F',' '$4=="Warning" {print "- `"$1":"$2"`: "$6}')
+ERRORS=$(echo "$SWIFTLINT_OUTPUT" | awk -F',' '$4=="Error" {print "- `"$1":"$2"`: "$6}')
```

Comment:
Won't this cause any issue if one of the columns in the CSV (like the warning or error message) contains a comma in it? Wouldn't use `json` reporter (+ `jq`) be safer?

I also just saw that `swiftlint reporters` lists `github-actions-logging` as one possible value (which could be useful if in the future we end up making Dangermattic be the one running swiftlint), or even directly as `markdown` (which seems like a better fit for Buildkite annotations?)

Reply:
I started with the `json` reporter and that would definitely better, but it turns out that the official SwiftLint image doesn't have `jq`. But good point on it having a comma, I'm gonna think a bit about it.

I've tried `markdown` before but I wanted to filter errors / warnings, and iirc it wasn't as simple as CSV (I can give it another try). I also wasn't a big fan of how it looked:
file | line | severity | reason | rule_id
--- | --- | --- | --- | ---
/Users/ian/dev/a8c/pocket-casts-ios/podcasts/CarPlaySceneDelegate.swift | 169 | :warning: | Trailing Whitespace: Lines should not have trailing whitespace | trailing_whitespace
/Users/ian/dev/a8c/pocket-casts-ios/podcasts/Settings.swift | 895 | :warning: | Trailing Whitespace: Lines should not have trailing whitespace | trailing_whitespace
/Users/ian/dev/a8c/pocket-casts-ios/podcasts/Onboarding/Models/PlusPurchaseModel.swift | 168 | :warning: | Trailing Whitespace: Lines should not have trailing whitespace | trailing_whitespace
/Users/ian/dev/a8c/pocket-casts-ios/podcasts/Onboarding/OnboardingFlow.swift | 26 | :warning: | Trailing Whitespace: Lines should not have trailing whitespace | trailing_whitespace
/Users/ian/dev/a8c/pocket-casts-ios/podcasts/SettingsViewController.swift | 182 | :warning: | Trailing Whitespace: Lines should not have trailing whitespace | trailing_whitespace

If we go to the SwiftLint via Danger, I don't think you can set a reporter as it uses the json one to parse the results and post back to Github.

Reply:
I think filtering the markdown output shouldn't be that hard, using ` | grep '| :warning: |'`?

Reply:
I've tweaked the code (475e4e97936e4b7b79996d0687e45dc55fb4b8ec) a bit to use yet another reporter, `relative-path`. It uses a structured message format that seems less prone to errors such as the comma one on CSV, with the advantage of also reporting a file path relative to root.